core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -3473,6 +3473,8 @@ private void handlePostUpdateRoleName(String roleName, String newRoleName, boole
                 if (internalRole && listener instanceof AbstractUserOperationEventListener) {
                     success = ((AbstractUserOperationEventListener) listener).doPostUpdateInternalRoleName(roleName,
                             newRoleName, this);
+                } else if (internalRole && !(listener instanceof AbstractUserOperationEventListener)) {
+                    success = true;
```

Comment:
I think, we need to consider a scenario where a custom UserOperationEventListener is using, which implements the UserOperationEventListener interface. The doPostUpdateInternalRoleName of the custom implementation will not be executed with these changes, so should change this as below. WDYT?
success = ((AbstractUserOperationEventListener) listener).doPostUpdateInternalRoleName(roleName, newRoleName, this);

Reply:
Here in UserOperationEventListener interface  doPostUpdateInternalRoleName method is not defined. Hence there is no chance ti implement doPostUpdateInternalRoleName() in the  custom UserOperationEventListener. So the only possible way to stop this silent failure is to make 'success =true' by force.

Reply:
Got it. It would be better to create a issue for this to add "doPostUpdateInternalRoleName" in UserOperationEventListener interface in next major version.